execution/src/posix/execution.cc

```diff
@@ -188,11 +189,38 @@ namespace leatherman { namespace execution {
                 }
                 pipe.buffer.resize(4096);
             }
+            if (input.first != -1) {
+                FD_SET(input.first, &set);
+                if (input.first > max) {
+                    max = input.first;
+                }
+            }
             if (max == -1) {
                 // All pipes closed; we're done
                 return;
             }
 
+            if (input.first != -1 && FD_ISSET(input.first, &set)) {
+                // Expecting data on input
+                auto count = write(input.first, input.second.c_str(), input.second.size());
```

Comment:
Even though this is a blocking write, a signal could still cause less than `size` bytes to be written here. We should loop until everything has been written.